sentry-spring/src/main/java/io/sentry/spring/webflux/SentryWebExceptionHandler.java

```diff
@@ -0,0 +1,47 @@
+package io.sentry.spring.webflux;
+
+import io.sentry.IHub;
+import io.sentry.SentryEvent;
+import io.sentry.SentryLevel;
+import io.sentry.exception.ExceptionMechanismException;
+import io.sentry.protocol.Mechanism;
+import io.sentry.util.Objects;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.core.annotation.Order;
+import org.springframework.web.server.ResponseStatusException;
+import org.springframework.web.server.ServerWebExchange;
+import org.springframework.web.server.WebExceptionHandler;
+import reactor.core.publisher.Mono;
+
+/** Handles unhandled exceptions in Spring WebFlux integration. */
+@Order(
+    -2) // the DefaultErrorWebExceptionHandler provided by Spring Boot for error handling is ordered
+// at -1
+@ApiStatus.Experimental
+public final class SentryWebExceptionHandler implements WebExceptionHandler {
+  private final @NotNull IHub hub;
+  private final @NotNull TransactionNameProvider transactionNameProvider;
+
+  public SentryWebExceptionHandler(final @NotNull IHub hub) {
+    this.hub = Objects.requireNonNull(hub, "hub is required");
+    this.transactionNameProvider = new TransactionNameProvider();
+  }
+
+  @Override
+  public @NotNull Mono<Void> handle(
+      final @NotNull ServerWebExchange serverWebExchange, final @NotNull Throwable ex) {
+    if (!(ex instanceof ResponseStatusException)) {
+      final Mechanism mechanism = new Mechanism();
+      mechanism.setType("SentryWebExceptionHandler");
+      mechanism.setHandled(false);
+      final Throwable throwable =
+          new ExceptionMechanismException(mechanism, ex, Thread.currentThread());
```

Comment:
more like a question, is `SentryWebExceptionHandler` always executed in the calling thread? otherwise marking the current thread will be wrong due to the usage of `Thread.currentThread()`.

Asking because of https://github.com/getsentry/sentry-java/issues/1524#issuecomment-859370797
in coroutines, the handler can get called in different threads

Reply:
I don't think there is a guarantee that its executed on the same thread. `thread` is a required property - what should be set there other than `currentThread` in this case?

Reply:
the side effect is that the `exception.setThreadId` will contain the id of the current thread, which could display the wrong stack trace in the sentry issues page if `SentryOptions#attachThreads` is enabled (all threads reported).
I'm wondering if we should make `ExceptionMechanismException#getThread` nullable, and add a new ctor to `ExceptionMechanismException` and just don't set the `threadId` if not available.

@bruno-garcia ?

Reply:
ping @bruno-garcia

Reply:
If we include Thread stack traces + Exception stack trace, the UI will display the Exception stack trace on the threads list (and ignore the Thread one) by matching its ID. So if we don't send an ID this could be broken.

Since this is asynchronous, the "thread" is less relevant here in terms of "This thread has done all the work up to this point". But it's OK to be "this thread did the work at the time of capture" which is what this is represented here.

So I think keeping as `Thread.currentThread()` is fine here and arguably a better approach than making it nullable given how Sentry works.

Reply:
yep I am just afraid that the thread id may be wrong, and we'd be hiding the stack trace of that thread since it's going to be showing the exception stack trace, I don't know if it's a big problem though.